model_analyzer/config/config.py

```diff
@@ -470,6 +470,74 @@ def _setup_logger(self):
                             "[%(filename)s:%(lineno)d] %(message)s",
                             datefmt="%Y-%m-%d %H:%M:%S")
 
+    def _autofill_values(self):
+        """
+        Fill in the implied or default
+        config values.
+        """
+
+        new_model_names = {}
+        for model in self.model_names:
+            new_model_names[model.model_name()] = {}
+            if not model.objectives():
+                if 'objectives' not in self._fields:
+                    raise TritonModelAnalyzerException(
+                        "Must specify top-level 'objectives' section in config."
+                        f"Objectives not specified under model : {model.model_name()}"
+                    )
+                new_model_names[
+                    model.model_name()]['objectives'] = self.objectives
+            if not model.constraints():
+                if 'constraints' not in self._fields:
+                    raise TritonModelAnalyzerException(
+                        "Must specify top-level 'constraints' section in config."
+                        f"Constraints not specified under model : {model.model_name()}"
+                    )
+                new_model_names[
+                    model.model_name()]['constraints'] = self.constraints
+            if not model.parameters():
+                if 'batch_sizes' not in self._fields or 'concurrency' not in self._fields:
```

Comment:
This condition is never True. batch_sizes and concurrency are the required field and must be specified.